### PR TITLE
Introduced top level factory

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
Includes fixing the web entry point so it actually includes routes.php :)

Preparation for adding logger needed by https://phabricator.wikimedia.org/T123049

### CODE REVIEW

File: src/TopLevelFactory.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		}
+
+		return $this->pdo;
+	}
+
+	public function newTestDeployer(): Deployer {
+		return new Deployer( new PdoReleaseRepository( $this->getPdo(), self::TEST_BRANCH_NAME ) );
+	}
+
+	public function newReleaseStateWriter(): ReleaseStateWriter {
```
Comment (gbirke) [inline src/TopLevelFactory.php]: Shouldn't this be `getReleaseStateWriter` for consistency?

`get` = "create one instance and always return the same one"  
`new` = "always return a new instance"

File: app/bootstrap.php [hunk 2]
```diff
@@ -1,17 +1,3 @@
...
-	return new \PDO( $app['dsn'] );
-};
-
-$app['release_state_writer'] = function ( $app ) {
-	return new ReleaseStateWriter( $app['db'] );
-};
-
-return $app;
\ No newline at end of file
+return new Silex\Application();
```
Comment (gbirke) [inline app/bootstrap.php:3]: Until we do some real bootstrapping (which won't happen now that we've got a TopLevelfactory) we can get rid of this file and just create `$app` in `index.php`

Comment (JeroenDeDauw) [reply app/bootstrap.php:3]: It's also needed in WebTestCase

File: src/TopLevelFactory.php (hunk 1 above)
Comment (JeroenDeDauw) [reply src/TopLevelFactory.php]: Done

**Pull request merged**